src/home/celebration/NftCelebration.tsx

```diff
@@ -0,0 +1,252 @@
+import GorhomBottomSheet, { BottomSheetBackdrop, BottomSheetView } from '@gorhom/bottom-sheet'
+import { BottomSheetDefaultBackdropProps } from '@gorhom/bottom-sheet/lib/typescript/components/bottomSheetBackdrop/types'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { HomeEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { BottomSheetRefType } from 'src/components/BottomSheet'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import { nftCelebrationDisplayed } from 'src/home/actions'
+import ConfettiCelebration from 'src/home/celebration/ConfettiCelebration'
+import { lastDisplayedNftCelebration } from 'src/home/selectors'
+import ImageErrorIcon from 'src/icons/ImageErrorIcon'
+import { nftsLoadingSelector, nftsWithMetadataSelector } from 'src/nfts/selectors'
+import { NftWithMetadata } from 'src/nfts/types'
+import { getDynamicConfigParams, getFeatureGate } from 'src/statsig'
+import { DynamicConfigs } from 'src/statsig/constants'
+import { StatsigDynamicConfigs, StatsigFeatureGates } from 'src/statsig/types'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { vibrateSuccess } from 'src/styles/hapticFeedback'
+import { Spacing } from 'src/styles/styles'
+
+export default function NftCelebration() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const insets = useSafeAreaInsets()
+  const insetsStyle = { paddingBottom: Math.max(insets.bottom, Spacing.Regular16) }
+
+  const bottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const [showAnimation, setShowAnimation] = useState(false)
+  const animationStartTime = useRef(0)
+
+  const [nftsLoaded, setNftsLoaded] = useState(false)
```

Comment:
yes, you are right: that trick wouldn't help if the fetch hadn't yet started. even worse, having misinterpreted the `nftsLoading` false value as _fetch is completed_ it will prevent the celebration from displaying.
 
my initial aim was to ensure that this bottom sheet doesn't reopen accidentally if the nfts fetch is restarted for some reason before we have finished the celebration, e.g. during the confetti animation. although it's not the case now, I wanted to ensure some future proof. uh, looks like this approach is a suboptimal way of doing it.

I think allowing a null value may not help, since the `nfts` list persists between launches. perhaps we can have a dedicated flag for displaying celebration right in the redux state, and perform all necessary checks within the saga. that way we can ensure that the fetch actually happened and succeeded. the celebration flag would be turned on after a successful fetch and turned off after a successful display. what do you think?

in case of initial fetch failure, I would prefer to leave the responsibility of initiating retries to the nfts fetching saga itself, rather than related components. would be happy to address it in a separate PR.